src/fileio/datafile.cxx

```diff
@@ -105,29 +97,14 @@ Datafile& Datafile::operator=(Datafile &&rhs) {
   f3d_arr      = rhs.f3d_arr;
   v2d_arr      = rhs.v2d_arr;
   v3d_arr      = rhs.v3d_arr;
-  if (filenamelen < rhs.filenamelen){
-    delete[] filename;
-    filenamelen=rhs.filenamelen;
-    filename=new char[filenamelen];
-  }
-  strncpy(filename,rhs.filename,filenamelen);
+  filename     = rhs.filename;
   return *this;
 }
 
-Datafile::~Datafile() {
-  if (filename != nullptr){
-    delete[] filename;
-    filename=nullptr;
-    filenamelen=0;
-  }
-}
+bool Datafile::openr(const std::string& format) {
 
-bool Datafile::openr(const char *format, ...) {
-  if(format == (const char*) NULL) 
-    throw BoutException("Datafile::open: No argument given for opening file!");
+  filename = format;
```

Comment:
I would prefer to directly change the argument of the function to `filename`, instead of `format` ...